src/PortfolioPlanning/Redux/Sagas/EpicTimelineSaga.ts

```diff
@@ -45,19 +45,19 @@ function* onShiftEpic(action: ActionsOfType<EpicTimelineActions, EpicTimelineAct
     yield effects.call(saveDatesToServer, epicId);
 }
 
-function* saveDatesToServer(epicId: number): SagaIterator {
+function* saveDatesToServer(epicId: number, defaultStartDate?: Date, defaultEndDate?: Date): SagaIterator {
     const epic: IEpic = yield effects.select(getEpicById, epicId);
 
     const doc: JsonPatchDocument = [
         {
             op: "add",
             path: "/fields/Microsoft.VSTS.Scheduling.StartDate",
-            value: epic.startDate
+            value: defaultStartDate || epic.startDate

```

Comment:
Shouldn't this follow the start date? If a default is provided then it will be used instead of the set start date.

Reply:
This method `saveDatesToServer` is called when setting dates from UI (you have implemented long back) and setting defaults date when adding a Epic that doesn't have dates (the code I have added).

I only want the defaultStateDate and defaultEndDate to be passed in the latter.

Also for your question, I couldn't do that because epic is undefined in the second case because it's not saved into state yet. SO I am basically counting on no default dates are passed in for case one.

I think it's kind of fragile?
I will add some comment and make some changes to this method.